### PR TITLE
Emails: Fix small 15px gap in thank you page for Professional Email

### PR DESCRIPTION
**Pull request by AllTerrainDeveloper**
#### Changes proposed in this Pull Request

Thank you page hero card has a small 15px gap between MasterBar and the Thank you page itself. The fix just adds into the equation the MasterBar height, so the gap is not prefixed and it's a formula:

`calc( -79px + var( --masterbar-height ) )`

In addition this Pull Request is unifying the two only styles that ThankYou component uses. So it requires less customisation when using this component.

#### Testing instructions

1. Go to Upgrades -> Emails
2. Buy or add a mailbox (Professional Email)
3. Check the thank you page is not showing an empty gap in the top side as per the below table.

To test the domain Thank you Page please, refer to this [Pull Request](https://github.com/Automattic/wp-calypso/pull/55675)

BEFORE | AFTER
------------ | -------------
image | image

#### Mobile
Email | Domain
------------ | -------------
thank you titan | domain you titan

#### Remarks

This issue is not replicable with old purchases, or reloading the site. In this case the order of stylesheet loading is different and makes not possible to replicate the issue.

Related to {1200182182542585-as-1201006404457937}

### CODE REVIEW

File: client/my-sites/email/titan-set-up-thank-you/style.scss [hunk 1]
```diff
@@ -6,7 +6,8 @@
 }
 
 .thank-you__container {
```
Comment (daledupreez) [inline client/my-sites/email/titan-set-up-thank-you/style.scss]: I think this is the wrong approach -- I think the root problem is that this CSS selector is currently too general, so it can override the core specifications for this selector. Given that this CSS should only be used when we're within the emails section, can we limit this CSS to `.is-section-email`? I think we should still check that other uses of the Thank You component continue to work, however, as this overly general CSS might have been interfering in other contexts before this.
```suggestion
.is-section-email .thank-you__container {
```

Comment (AllTerrainDeveloper) [reply client/my-sites/email/titan-set-up-thank-you/style.scss]: In this case, I wonder if we should also remove `.thank-you__container` class, as we can completely distinguish this HTML component already only by `.is-section-email`

Comment (AllTerrainDeveloper) [reply client/my-sites/email/titan-set-up-thank-you/style.scss]: Plus I think, other instances of the `thank-you` components are not going to be affected, because this is the `titan-set-up-thank-you` component 🙂, which creates `thank-you` component with modified CSS.

Comment (AllTerrainDeveloper) [reply client/my-sites/email/titan-set-up-thank-you/style.scss]: And after rethinking a bit, it might be even better being a bit more descriptive about the class name, like `titan-set-up-thank-you__container` instead of `.is-section-email`, don't you think so?

**Comment by stephanethomas** [comment]
I'm a bit lost. I had a look to production, and didn't notice any display issue with the master bar:

screenshot

I've tested with Chrome on Windows 10.

**Comment by stephanethomas** [comment]
After having a look to the initial discussion on Slack, I was able to reproduce it when purchasing a new mailbox. This issue doesn't happen when setting up an unused mailbox though:

Setting Up Mailbox | Purchasing Mailbox
------ | -----
screenshot | screenshot

@saygunnyc: we display the sidebar in one case but not the other, should we address that so those pages are consistent?

**Comment by stephanethomas** [comment]
Something that was mentioned in the Asana board (and unfortunately not mentioned in this pull request) is that this gap only appears the first time the `Thank You` page is rendered - it will disappear when reloading the page. I believe this is because styles are loaded in a different order in those two scenarios, as you can see here:

Initial | Reload
------ | ----
initial | reload

Rather than duplicating a rule to override the margin, maybe we should see how to make [those CSS rules](https://github.com/Automattic/wp-calypso/blob/349efb1c0b3180ba875fefc669e3b392567ab6b3/client/components/thank-you/index.tsx#L18-L21) specific enough so that they are no longer dependent on the loading order?

As an aside, I've noticed that the `Thank You` page for domains also has some [specific rules](https://github.com/Automattic/wp-calypso/blob/349efb1c0b3180ba875fefc669e3b392567ab6b3/client/my-sites/checkout/checkout-thank-you/domains/style.scss#L15-L26) related to that issue. Is there a way we could address that issue for both flows without duplicating rules?

**Comment by AllTerrainDeveloper** [comment]
> Something that was mentioned in the Asana board (and unfortunately not mentioned in this pull request) is that this gap only appears the first time the `Thank You` page is rendered - it will disappear when reloading the page. I believe this is because styles are loaded in a different order in those two scenarios, as you can see here:
> 
> Initial	Reload
> initial	reload
> Rather than duplicating a rule to override the margin, maybe we should see how to make [those CSS rules](https://github.com/Automattic/wp-calypso/blob/349efb1c0b3180ba875fefc669e3b392567ab6b3/client/components/thank-you/index.tsx#L18-L21) specific enough so that they are no longer dependent on the loading order?
> 
> As an aside, I've noticed that the `Thank You` page for domains also has some [specific rules](https://github.com/Automattic/wp-calypso/blob/349efb1c0b3180ba875fefc669e3b392567ab6b3/client/my-sites/checkout/checkout-thank-you/domains/style.scss#L15-L26) related to that issue. Is there a way we could address that issue for both flows without duplicating rules?

Sorry, I am adding the extra bit of not replicable when reloading the site.

**Comment by saygunnyc** [comment]
> @saygunnyc: we display the sidebar in one case but not the other, should we address that so those pages are consistent?

I think removing the sidebar should act in our favor in terms of converting the user into their Inbox. Fewer options to click, fewer distractions from the main CTA. Just re-checked the original designs over this Figma – mZXo4ZagpIMWtu526c17W5-fi-1960%3A208, they also do not feature a sidebar however the context is a bit different once setting up a mailbox.

Do you guys think there's any value in having the sidebar? Anything I might be missing?

**Comment by AllTerrainDeveloper** [comment]
@daledupreez I've unified as much as possible the styling so we don't need so much customisation to use this thank you component.

Also, we haven't talk yet about @saygunnyc 's concern:
> Do you guys think there's any value in having the sidebar? Anything I might be missing?

File: client/components/thank-you/style.scss [hunk 2]
```diff
@@ -59,3 +41,128 @@
...
+		&-title {
+			line-height: 40px;
+		}
+
+		&-subtitle {
+			padding-top: 4px;
+			margin-bottom: 0;
+		}
+	}
+}
```
Comment (daledupreez) [inline client/components/thank-you/style.scss]: These are duplicated below in the `.thank-you__container` section on L68 and in the `&__header` section on L83.

File: client/components/thank-you/style.scss [hunk 3]
```diff
@@ -59,3 +41,128 @@
...
+
+	&__container-header {
+		width: auto;
+		background-color: var( --studio-blue-50 );
+		padding: 0 20px 35px;
+		box-sizing: border-box;
+		text-align: center;
+		height: auto;
+		flex-direction: column;
+		align-items: center;
```
Comment (daledupreez) [inline client/components/thank-you/style.scss]: Nit: can these be in alphabetical order, please?

File: client/components/thank-you/style.scss [hunk 4]
```diff
@@ -1,45 +1,27 @@
...
-	div > p {
-		margin-bottom: 12px;
-		@include break-mobile {
-			margin-bottom: 24px;
-		}
-	}
+.thank-you__step-cta > a.button:not( .is-primary ) {
+	border-color: var( --studio-gray-10 );
+	color: var( --studio-gray-80 );
+	background-color: var( --studio-white );
```
Comment (daledupreez) [inline client/components/thank-you/style.scss:18]: Is this change intentional? It means we're ignoring the colours for the current theme, which may or may not be part of the changes we want to make here.

File: client/components/thank-you/style.scss [hunk 5]
```diff
@@ -59,3 +41,128 @@
...
+			flex-direction: row;
+		}
+
+		div > p {
+			@include break-mobile {
+				margin-bottom: 0;
+			}
+		}
+
+		&-cta {
```
Comment (daledupreez) [inline client/components/thank-you/style.scss:100]: (Strong) personal preference: I'd rather have these selectors not use `&` as I find it harder to work out what CSS class/element we're targeting (especially when they're nested like this), and it also makes it harder for me to find the CSS if I'm coming from the component.
As an example, I need to look __really__ closely at lines 81 and 112 to know that this resolves to `.thank-you__step-cta`, and I would not be able to find this selector without some really creative searching in my IDE if I am starting from the component.

File: client/components/thank-you/style.scss [hunk 6]
```diff
@@ -59,3 +41,128 @@
...
+	}
+
+	&__body {
+		@include break-zoomed-in {
+			margin-top: 32px;
+		}
+
+		@include break-small {
+			margin-top: 50px;
+		}
```
Comment (daledupreez) [inline client/components/thank-you/style.scss:74]: Are these reacting to the masterbar height at these breakpoints? Can we use a variable instead? I am really worried that this will be fragile in the long term, most likely if we shift the breakpoints for the masterbar.

File: client/components/thank-you/style.scss [hunk 7]
```diff
@@ -1,45 +1,27 @@
...
-		max-width: 150px;
-	}
+.is-section-checkout-thank-you {
+	background: var( --studio-white );
 }
 
-.thank-you__container-header {
-	flex-direction: column;
-	text-align: center;
-	align-items: center;
```
Comment (daledupreez) [inline client/components/thank-you/style.scss:15]: Are these no longer needed?

**Review by daledupreez — COMMENTED** [review]
I haven't fully tested yet, but want to get these comments in before I get that wrapped.

File: client/components/thank-you/style.scss [hunk 8]
```diff
@@ -59,3 +41,128 @@
...
+
+	margin-top: calc( -79px + var( --masterbar-height ) );
+
+	@include breakpoint-deprecated( '<960px' ) {
+		margin-top: calc( -71px + var( --masterbar-height ) );
+	}
+
+	@include breakpoint-deprecated( '<660px' ) {
+		margin-top: -1px;
+	}
```
Comment (daledupreez) [inline client/components/thank-you/style.scss]: Not for this PR, but the `-79px`/`-71px`/`-1px` magic numbers along with the respective breakpoints definitely need to be shifted to a variable.

OK - I created #56654 for this. We have multiple places in Calypso that are doing weird things with these hard-coded values and breakpoints, and they are all liable to break in future.

File: client/components/thank-you/style.scss (hunk 5 above)
Comment (daledupreez) [reply client/components/thank-you/style.scss:100]: Let's have a discussion about two questions:
1. Is it worth making these values easier to find and understand? (I now realise they came from the domain-specific changes)
2. Does it make more sense to refactor them in a follow-up PR?

I am leaning to "Yes" answers for both -- we should do this, but probably in a follow-up PR.

**Review by daledupreez — CHANGES_REQUESTED** [review]
Functionally this is working as expected, but I think we need to remove the code duplication I identified earlier on.

**Comment by AllTerrainDeveloper** [comment]
I completely agree with you about the SCSS syntax that allow "rule" composition. I just took it from domains styling, with the intention of leave everything untouched and avoid regression. And I completely agree to do a refactor in a follow up pull request. Having said this, I'll remove the duplicated styling! 🙂 

PS: Honestly I didn't know it was possible to do that "rule composition" it's been a surprise.

File: client/components/thank-you/style.scss (hunk 4 above)
Comment (AllTerrainDeveloper) [reply client/components/thank-you/style.scss:18]: I did it intentional, as margin all the styles, our thank you page ended like this:
image

While it should be like this:
image

File: client/my-sites/email/titan-set-up-thank-you/style.scss [hunk 9]
```diff
@@ -4,10 +4,3 @@
 .titan-set-up-thank-you__icon-external {
 	margin-left: 4px;
 }
-
-.thank-you__container {
-	margin: -32px -32px -33px -34px;
-	@media ( max-width: 660px ) {
-		padding: 0 32px 34px 34px;
-	}
-}
```
Comment (olaseni) [inline client/my-sites/email/titan-set-up-thank-you/style.scss:13]: Unfortunately removing these rules breaks an existing view in the `<TitanSetUpThankYou/>` component:

`/email/<domain>/titan/set-up-mailbox/thank-you/<site>`.

Visuals:

Screenshot 2021-10-09 at 1 47 05 PM

**Review by olaseni — CHANGES_REQUESTED** [review]
Might it be better to split this task into different units? To reduce the error surface area, I think you should fix the 15px issue ONLY in this PR.

**Comment by AllTerrainDeveloper** [comment]
I’m open to any change, initially it was only a fix for the 15px gap, but with the initial review it got a bigger scope.

**Comment by AllTerrainDeveloper** [comment]
Closing to open different branch with only requested changes
